ttnn/cpp/ttnn/operations/matmul/device/matmul_op.hpp

```diff
@@ -225,6 +225,36 @@ MatmulProgramConfig create_matmul_program_config(
     std::optional<UnaryWithParam> fused_activation,
     std::optional<const DeviceComputeKernelConfig> compute_kernel_config);
 
+inline Matmul create_matmul_struct(
```

Comment:
Please do not make this inline and move the code into the cpp file.

I'm in the process of doing that for the other methods in the hpp file. Hopefully that will help with the compile time problem we're having.

Is the reason for you to create a matmul struct to re-use validate, compute_output_shapes, and create_output tensors?

Why do you want to call matmul's validate, which has different requirements depending on which matmul program config is chosen?

Reply:
The reason we're trying to reuse the valid, compute_output_shapes, and create_output tensors from the matmul code is because the fused ops (all_gather and matmul in this case), should still carry out mostly as they would in a non-fused manner. That means that the matmul checks should hold, as well as the shapes of the output tensor. If we need additional checks, they would be added on top of the base op's checks.

Reply:
Ok. Thanks.